src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int START_VALUE = 0;
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] separateDataFromFile = readFromFile(fromFileName);
+        int[] reportValues = calculateReport(separateDataFromFile);
+        writeInfoToFile(reportValues, toFileName);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+                new FileReader(fromFileName))) {
+            StringBuilder builder = new StringBuilder();
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                builder.append(line).append(COMMA);
+                line = bufferedReader.readLine();
+            }
+            return builder.toString().split(COMMA);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file " + fromFileName + e.getMessage());
+        }
+    }
+
+    private int[] calculateReport(String[] dataLines) {
+        int supplyValue = START_VALUE;
+        int buyValue = START_VALUE;
+        for (int i = 0; i < dataLines.length; i++) {
+            if (dataLines[i].equals(SUPPLY)) {
+                supplyValue += Integer.parseInt(dataLines[i + 1]);
+            } else if (dataLines[i].equals(BUY)) {
+                buyValue += Integer.parseInt(dataLines[i + 1]);
+            }
+        }
+        return new int[]{supplyValue, buyValue};
+    }
 
+    private void writeInfoToFile(int[] calculateReport, String toFile) {
+        StringBuilder builder = new StringBuilder();
```

Comment:
bad naming, because it doesn't tell us anything about what we expect to store in this variable - don't name List list, [] array etc, name is an opportunity to make code easier to read :)